api/src/convert/signed_contingent_input.rs

```diff
@@ -0,0 +1,159 @@
+    use mc_fog_report_validation_test_utils::MockFogResolver;
+    use mc_transaction_builder::{
+        test_utils::get_input_credentials, EmptyMemoBuilder, ReservedSubaddresses,
+        SignedContingentInputBuilder,
+    };
+    use mc_transaction_core::{
+        constants::MILLIMOB_TO_PICOMOB, tokens::Mob, Amount, BlockVersion, Token, TokenId,
+    };
+    use protobuf::Message;
+    use rand::{rngs::StdRng, SeedableRng};
+
+    #[test]
+    /// SignedContingentInput --> external::SignedContingentInput -->
+    /// SignedContingentInput should be the identity function
+    fn test_convert_signed_contingent_input() {
+        let mut rng: StdRng = SeedableRng::from_seed([1u8; 32]);
+        let charlie = AccountKey::random(&mut rng);
+
+        let token2 = TokenId::from(2);
+
+        let fpr = MockFogResolver::default();
+
+        // Charlie makes a signed contingent input, offering 1000 token2's for 1 MOB
+        let input_credentials = get_input_credentials(
+            BlockVersion::MAX,
+            Amount::new(1000, token2),
+            &charlie,
+            &fpr,
+            &mut rng,
+        );
+        let proofs = input_credentials.membership_proofs.clone();
+        let mut sci_builder = SignedContingentInputBuilder::new(
+            BlockVersion::MAX,
+            input_credentials,
+            fpr.clone(),
+            EmptyMemoBuilder::default(),
+        )
+        .unwrap();
+
+        // Originator requests an output worth 1MOB destined to themselves
+        sci_builder
+            .add_partial_fill_output(
+                Amount::new(1000 * MILLIMOB_TO_PICOMOB, Mob::ID),
+                &charlie.default_subaddress(),
+                &mut rng,
+            )
+            .unwrap();
+
+        // Change amount matches the input value
+        sci_builder
+            .add_partial_fill_change_output(
+                Amount::new(1000, token2),
+                &ReservedSubaddresses::from(&charlie),
+                &mut rng,
+            )
+            .unwrap();
+        let mut sci = sci_builder.build(&NoKeysRingSigner {}, &mut rng).unwrap();
+
+        sci.tx_in.proofs = proofs;
+        sci.tx_out_global_indices = vec![1, 2, 3, 4];
+
+        // decode(encode(sci)) should be the identity function.
+        {
+            let bytes = mc_util_serial::encode(&sci);
+            let recovered_sci = mc_util_serial::decode(&bytes).unwrap();
+            assert_eq!(sci, recovered_sci);
+        }
+
+        // Converting mc_transaction_extra::SignedContingentInput ->
+        // external::SignedContingentInput ->
+        // mc_transaction_extra::SignedContingentInput should be the identity
+        // function.
+        {
+            let external_sci: external::SignedContingentInput =
+                external::SignedContingentInput::from(&sci);
+            let recovered_sci: SignedContingentInput =
+                SignedContingentInput::try_from(&external_sci).unwrap();
```

Comment:
:upside_down_face:  take it or leave it
```suggestion
            let external_sci = external::SignedContingentInput::from(&sci);
            let recovered_sci = SignedContingentInput::try_from(&external_sci).unwrap();
```